connaisseur/crypto.py

```diff
@@ -10,11 +12,26 @@ def verify_signature(public_base64: str, signature_base64: str, message: str):
 
     Raises ValidationError if unsuccessful.
     """
-    public = base64.b64decode(public_base64)
-    pub_key = ecdsa.VerifyingKey.from_der(public)
+    pub_key = decode_and_verify_ecdsa_key(public_base64)
 
     signature = base64.b64decode(signature_base64)
 
     msg_bytes = bytearray(message, "utf-8")
 
     return pub_key.verify(signature, msg_bytes, hashfunc=hashlib.sha256)
+
+
+def decode_and_verify_ecdsa_key(public_base64: str):
+    """
+    Verifies that the provided public key in base64 encoding qualifies as a
+    proper ecdsa key and throws if not.
+    """
+    public = base64.b64decode(public_base64)
+    try:
+        pubkey = ecdsa.VerifyingKey.from_der(public)
+    except ecdsa.keys.MalformedPointError as err:
```

Comment:
Is there a reason for only catching this kind of Error? I encountered a `ecdsa.der.UnexpectedDER` error - wouldn't we want to catch that too?

Reply:
done. Now catching all.